testnet/stacks-node/src/tests/epoch_205.rs

```diff
@@ -913,3 +916,292 @@ fn test_cost_limit_switch_version205() {
+                )
+            )
+        })
+        .collect();
+
+    let (mut conf, miner_account) = neon_integration_test_conf();
+
+    for spender_addr in spender_addrs.iter() {
+        conf.initial_balances.push(InitialBalance {
+            address: spender_addr.clone(),
+            amount: 10492300000,
+        });
+    }
+
+    conf.node.mine_microblocks = true;
+    conf.node.wait_time_for_microblocks = 1000;
+    conf.node.microblock_frequency = 1000;
+    conf.node.max_microblocks = 65536;
+    conf.burnchain.max_rbf = 1000000;
+
+    conf.miner.min_tx_fee = 1;
+    conf.miner.first_attempt_time_ms = i64::max_value() as u64;
+    conf.miner.subsequent_attempt_time_ms = i64::max_value() as u64;
+
+    conf.burnchain.epochs = Some(vec![
+        StacksEpoch {
+            epoch_id: StacksEpochId::Epoch20,
+            start_height: 0,
+            end_height: 208,
+            block_limit: ExecutionCost {
+                write_length: 15000000,
+                write_count: 7750,
+                read_length: 100000000,
+                read_count: 7750,
+                runtime: 5000000000,
+            },
+            network_epoch: PEER_VERSION_EPOCH_2_0,
+        },
+        StacksEpoch {
+            epoch_id: StacksEpochId::Epoch2_05,
+            start_height: 208,
+            end_height: 9223372036854775807,
+            block_limit: ExecutionCost {
+                write_length: 15000000 * 2,
+                write_count: 7750 * 2,
+                read_length: 100000000 * 2,
+                read_count: 7750 * 2,
+                runtime: 5000000000 * 2,
+            },
+            network_epoch: PEER_VERSION_EPOCH_2_05,
+        },
+    ]);
+
+    test_observer::spawn();
+    conf.events_observers.push(EventObserverConfig {
+        endpoint: format!("localhost:{}", test_observer::EVENT_OBSERVER_PORT),
+        events_keys: vec![EventKeyType::AnyEvent],
+    });
+
+    let mut btcd_controller = BitcoinCoreController::new(conf.clone());
+    btcd_controller
+        .start_bitcoind()
+        .map_err(|_e| ())
+        .expect("Failed starting bitcoind");
+
+    let mut btc_regtest_controller = BitcoinRegtestController::new(conf.clone(), None);
+    let http_origin = format!("http://{}", &conf.node.rpc_bind);
+
+    btc_regtest_controller.bootstrap_chain(201);
+
+    eprintln!("Chain bootstrapped...");
+
+    let mut run_loop = neon::RunLoop::new(conf);
+    let blocks_processed = run_loop.get_blocks_processed_arc();
+    let microblocks_processed = run_loop.get_microblocks_processed_arc();
+
+    let channel = run_loop.get_coordinator_channel().unwrap();
+
+    thread::spawn(move || run_loop.start(None, 0));
+
+    // give the run loop some time to start up!
+    wait_for_runloop(&blocks_processed);
+
+    // first block wakes up the run loop
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    // first block will hold our VRF registration
```

Comment:
we have two "first block"s in the comments

you can optionally use `run_until_burnchain_height` in `testnet/stacks-node/src/tests/epoch_205.rs`. this takes a "burnchain height" argument, so if the tests passes the reader will be able to know exactly which block we're on.

the numbers then can't be out of date because they'll be part of the test

Reply:
Fixed